client/galaxy/scripts/components/History/model/DatasetCollection/DatasetCollection.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Our API is very inconsistent which makes re-using components difficult.
```

Comment:
Can you create an issue for what you'd like to see different - or maybe just a comment on https://github.com/galaxyproject/galaxy/issues/6306 or https://github.com/galaxyproject/galaxy/issues/1810 - which ever seems more appropriate?

Is this a formatting issue or a fundamental issue (attribute names don't match up) with the models (frustrating that the history panel uses the element identifier in lieu of the dataset name when element is in a collection)?

Reply:
@jmchilton 

I will do that. There are a couple problems, I'll gather up all my whining into one place for you.

I think the biggest issue is that a dataset that is returned in a collection is not the same format as a dataset that exists on a tree, and it is not possible to configure the fields that are returned in a consistent way.

The tree format of the dataset collection return is particularly clumsy in my opinion.

And if I recall one of the things that's making this thing very hard to scale for large collections is the inability to page (limit/offset) through the contents of a dataset collection.

I'll do a more thorough summary in an issue as you have suggested though, thanks!